lib/deaf_grandma.rb

```diff
@@ -19,9 +20,16 @@ def run!
 
 
   def speak(input)
-
-    #Implement your code here <<<<<<<<<
-
+    if input != input.upcase
+      "SPEAK UP SONNY!"
+    elsif input == "BYE"
+      @bye_counter += 1 
+      if @bye_counter == 3
+        abort("SEE YOU LATER SONNY!")
+      end 
+    elsif input == input.upcase
+      "NOT SINCE 1964!"
+    end 
```

Comment:
You might be able to use a `case` statement instead of the `if/elsif` statement.